edb/pgsql/compiler/relgen.py

```diff
@@ -2401,6 +2444,17 @@ def _process_set_func_with_ordinality(
             pathctx.put_path_value_var(
                 ctx.rel, element.path_id, element.val, env=ctx.env)
 
+    if (output.in_serialization_ctx(ctx) and ir_set.shape
+            and not ctx.env.ignore_object_shapes):
+        ensure_source_rvar(ir_set, ctx.rel, ctx=ctx)
+        exprcomp._compile_shape(ir_set, ir_set.shape, ctx=ctx)
+
+    var = pathctx.maybe_get_path_var(
+        ctx.rel, ir_set.path_id, aspect='serialized', env=ctx.env)
```

Comment:
What's going on here?

Reply:
Adding a comment:
```
    # If there is a shape specified on the argument to enumerate, we need
    # to compile it here manually, since we are skipping the normal
    # code path for it.
```